Week1 [STEP 3] 고도

gxdxt opened the pull request:
@DragonTigerXYZ 

안녕하세요 `Alan(앨런)` !
STEP 3 에 대한 PR입니다.

이번엔 TODO를 놓치지 않으려고 꼼꼼히 읽어보고, PR 보냅니다!

***

고민은 역시 `누드 빼빼로`에 대한 처리였는데요,
해당 예시덕분에, 예외처리를 해야 했습니다.
도저히 `누드 빼빼로`를 포함할 수 있는 함수가 생각이 나질 않더라구요.

그 외, TODO에 `새로운 파일을 생성하여` 라는 문구 때문에, main에 함수 구현을 하지 않고
호출만 하는 형식으로 작성하였습니다!

감사합니다 😄

Commit:
feat : Create the func for drawing the Pepero
 - create the func for drawing the body of Pepero
 - create the func for drawing the stick of Pepero
 - create the func for showing the paremeters which is given

Commit:
Merge branch 'develop' into main

Commit:
docs : Add the chartflow for STEP 3

Commit:
style : Change styles of code followed by API Design Guidelines

DragonTigerXYZ commented on CodeStarterCamp_Week1/drawingPPR.swift:
```diff
@@ -31,33 +31,33 @@ func drawStick(heightOfStick: Int) {
-var mainPPR_odd: String = ""
-var mainPPR_even: String = ""
+var bodyOdd: String = ""
+var bodyEven: String = ""
```
Swift에서 코드 작성시 정답은 아니지만 보통 변수에서 함수 순으로 정의합니다📝

DragonTigerXYZ commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,5 @@
 // 빼빼로 그리기
-drawTotal(heightOfStick: 10, bodyOfMain: "***", toppingOfMain: "^", heightOfPPR: 20)
+makeTotal(heightOfStick: 5, peperoBody: " |0|", topping: "", height: 7)
 
```
- 콘솔창에 나오는 정보 중 `길이`와 `막대길이`와 makeTotal 입력값의 순서가 다른데 의도가 있을까요?🤔

> 누드 빼빼로 좌측에는 빈칸이 하나 추가되어 있습니다. 라는 부분 때문에
좌측에 빈칸이 있는 조건 하에 '|'를 기준으로 split하는 코드를 입력해 넣었기에

빈칸 조건을 구현하기 위해 `peperoBody` 인자값에  " |0|"을 입력한 걸로 보이는데,
실행예시 1의 일반빼빼로도 몸통 인자값에 " ***" 이렇게 입력해야할까요?🤔

인자값에는 값만 넣고 조건은 실제 동작하는 함수에서 처리되는건 어떨까요?

gxdxt replied:
말씀해주신대로, 인자값에는 값만 넣고 조건에서 추가되도록 변경하였습니다 🔥

DragonTigerXYZ commented on CodeStarterCamp_Week1/drawingPPR.swift:
```diff
@@ -31,33 +31,33 @@ func drawStick(heightOfStick: Int) {
-        mainPPR_even = " " + "|" + bodyOfMainArray[1] + "|" + topping
+    if (peperoBody.contains("|")) {
+        let peperoBodyArray: Array = peperoBody.split(separator: "|")
+        bodyOdd = realTopping + "|" + peperoBodyArray[0] + "|" + " "
```
bodyOdd의 마지막에 " " 공백한 칸은 없어도 되지 않을까요?

DragonTigerXYZ commented on CodeStarterCamp_Week1/drawingPPR.swift:
```diff
@@ -31,33 +31,33 @@ func drawStick(heightOfStick: Int) {
         print(" |", "| ");
     }
 }
```
스틱의 끝부분 "| "에서  공백한 칸은 없어도 되지 않을까요?

gxdxt replied:
좌우 대칭을 맞춰줘야 깔끔해보일 것 같아서 불필요하더라도 공백 칸을 추가하였습니다 😢

DragonTigerXYZ commented on CodeStarterCamp_Week1/drawingPPR.swift:
```diff
@@ -10,18 +10,18 @@ import Foundation
-    drawMainPPR(bodyOfMain: bodyOfMain, toppingOfMain: toppingOfMain)
-    drawBodyOfPPR(heightOfPPR: heightOfPPR)
+func makeTotal(heightOfStick: Int, peperoBody: String, topping: String, height: Int) {
+    makePeperoInfo(heightOfStick: heightOfStick, peperoBody: peperoBody, topping: topping, height: height)
```
`makeTotal`이 메서드명만 봤을땐, 무슨일을 하는지 잘 모르겠어요😭

gxdxt replied:
앗,,, 해당 부분 변경 완료하였습니다! 말씀해주신대로 직관적으로 이해되도록 makeFullPepero로 변경하였습니다!

Commit:
refactor : Change the location of the variables to the top, etc
 - Change the location of the variables to the top
 - Change the name of the func called makeTotal
 - Change the seq of the parameters
 - Remove the unnecessary elements in String var

DragonTigerXYZ commented on CodeStarterCamp_Week1/drawingPPR.swift:
```diff
@@ -7,10 +7,12 @@
 
-func makeTotal(heightOfStick: Int, peperoBody: String, topping: String, height: Int) {
+func makeFullPepero(peperoBody: String, topping: String, height: Int, heightOfStick: Int) {
     makePeperoInfo(heightOfStick: heightOfStick, peperoBody: peperoBody, topping: topping, height: height)
```
`makeFullPepero` 함수의 매개변수 선언 순서는, `makePeperoInfo`의 출력순서와 다른데 하나로 맞춰주는게 좀 더 보기좋을거같아요!😅

DragonTigerXYZ commented on CodeStarterCamp_Week1/drawingPPR.swift:
```diff
@@ -42,8 +42,9 @@ func drawPeperoBody(peperoBody: String, topping: String) {
-        let peperoBodyArray: Array = peperoBody.split(separator: "|")
-        bodyOdd = realTopping + "|" + peperoBodyArray[0] + "|" + " "
+        var addBlankOnPeperoBody = " " + peperoBody
+        let peperoBodyArray: Array = addBlankOnPeperoBody.split(separator: "|")
```
`addBlankOnPeperoBody`를 var로 선언하신 이유에 대해 들어볼 수 있을까요?

gxdxt replied:
처음 코드 작성 당시에는, 'peperoBody 값이 변하니 상수로 선언하면 안되겠다!' 라고 생각했는데 지금 보니 해당 함수가 실행될 때는 변할 일이 없는 것 같아요! 상수로 변경하도록 하겠습니다 ! 😮

Commit:
refactor : Change the order of the parameters for matching with the func's parameters

Commit:
refactor : Change variable to constant

Commit:
refactor : Delete all comments and change the Namings
